### ISSUE TITLE
__dict_args__ not working as expected

### ISSUE THREAD

**Issue by adm271828** [dgilland/sqlservice#26]
Hello Derrick,

I suggest the last test in `_get_dict_adapter` to be replaced with:

```python
    try:
        if isinstance(value, cls):
            adapter = _adapter
            break
    except TypeError: pass
```
This will avoid an exception when `__args_dict__` does not provide an adapter for every key, and allow the default adapter to be returned.

```python
class User(Model):
    __tablename__ = 'user'

    __dict_args__ = {
        'adapters': { 'name': lambda v, *_: 'XXX_' + v + '_XXX', }
    }
    
    id = Column(Integer(), primary_key=True)
    name = Column(String(100))
    email = Column(String(100))
```
The following works because `email` is `None`:

```python
In[34]: User(name='xxx').to_dict()
Out[34]: {'name': 'XXX_xxx_XXX'}
```

Otherwise, when serializing `email`, the test ends up with `cls` being a string (`'name'`) and not a type:

```python
In[35]: User(name='xxx',email='abc').to_dict()
Traceback (most recent call last):

  File "<ipython-input-35-a667e6c41429>", line 1, in <module>
    User(name='xxx',email='abc').to_dict()

  File "C:\Tools\Continuum\Anaconda3\lib\site-packages\sqlservice\model.py", line 215, in to_dict
    class_registry)

  File "C:\Tools\Continuum\Anaconda3\lib\site-packages\sqlservice\model.py", line 356, in _get_dict_adapter
    if isinstance(value, cls):

TypeError: isinstance() arg 2 must be a type or tuple of types
```

Moreover the documentation says the adapter can be a function that accepts up to three arguments. It seems to me this only work if it accepts exactly three argument (otherwise raise a TypeError and complains it was given 3 arguments). 

And wouldn't it be a nice feature to prevent a given field's serialization if the adapter in `__dict_args__` is `None`? For instance:

```python
    __dict_args__ = {
        'adapters': { 'my-secret-field': None }
    }
```

BTW very nice package. I came to it when trying to reuse a model that was developped inside a Flask application...

Best regards,

Antoine

**Comment by adm271828** [comment]
I see at least 2 use cases:

- using `__to_dict__()` as a serializer whose result will be sent in response to a REST query. In which case, a control over fieds that are not exposed might be needed
- extracting records from a database with a neutral format as fixtures to be reinjected into another database that share the same model. In which case, I would like autogenerated integer IDs not to be exposed, because they are not part of object identity, and might already be used in the destination database.

**Comment by dgilland** [comment]
I'll go ahead and have this feature added. I might not be able to get to until the weekend so if you wanted to take a stab at a PR, that'd be fine by me. Otherwise, I'll try to get it done soon.

**Comment by dgilland** [comment]
I've implemented the `None` adapter support to exclude fields from dict serialization in 97f80f526449839f488663d29ebe04d4220c3dd3 and released in `v1.1.3`: https://pypi.org/project/sqlservice/1.1.3/

Thanks again for the feature idea and bug report!
